torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +65,128 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: Optional[torch.dtype] = None
+    quant_min_lower_bound: Optional[Union[int, float]] = None
+    quant_max_upper_bound: Optional[Union[int, float]] = None
+    scale_min_lower_bound: Optional[Union[int, float]] = None
+    scale_max_upper_bound: Optional[Union[int, float]] = None
+
+
 @dataclass
 class DTypeConfig:
     """
     Config for the set of supported input/output activation, weight, and bias data types for the
     patterns defined in :class:`~torch.ao.quantization.backend_config.BackendConfig`.
+
+    Example usage::
+
+    >>> dtype_config1 = DTypeConfig(
+    ...     input_dtype=torch.quint8,
+    ...     output_dtype=torch.quint8,
+    ...     weight_dtype=torch.qint8,
+    ...     bias_dtype=torch.float)
+
+    >>> dtype_config2 = DTypeConfig(
+    ...     input_dtype=DTypeWithConstraints(
+    ...         dtype=torch.quint8,
+    ...         quant_min_lower_bound=0,
+    ...         quant_max_upper_bound=255,
+    ...     ),
+    ...     output_dtype=DTypeWithConstraints(
+    ...         dtype=torch.quint8,
+    ...         quant_min_lower_bound=0,
+    ...         quant_max_upper_bound=255,
+    ...     ),
+    ...     weight_dtype=DTypeWithConstraints(
+    ...         dtype=torch.qint8,
+    ...         quant_min_lower_bound=-128,
+    ...         quant_max_upper_bound=127,
+    ...     ),
+    ...     bias_dtype=torch.float)
+
+    >>> dtype_config1.input_dtype
+    torch.quint8
+
+    >>> dtype_config2.input_dtype
+    torch.quint8
+
+    >>> dtype_config2.input_dtype_with_constraints
+    DTypeWithConstraints(dtype=torch.quint8, quant_min_lower_bound=0, quant_max_upper_bound=255,
+                         scale_min_lower_bound=None, scale_max_upper_bound=None)
     """
-    input_dtype: Optional[torch.dtype] = None
-    output_dtype: Optional[torch.dtype] = None
-    weight_dtype: Optional[torch.dtype] = None
-    bias_dtype: Optional[torch.dtype] = None
-    is_dynamic: Optional[bool] = None
+    input_dtype_with_constraints: DTypeWithConstraints
+    output_dtype_with_constraints: DTypeWithConstraints
+    weight_dtype_with_constraints: DTypeWithConstraints
+    bias_dtype: Optional[torch.dtype]
+    is_dynamic: Optional[bool]
+
+    def __init__(
+        self,
+        input_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        output_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        weight_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        bias_dtype: Optional[torch.dtype] = None,
+        is_dynamic: Optional[bool] = None,
+    ):
+        if input_dtype is None:
+            input_dtype = DTypeWithConstraints()
```

Comment:
nit: update `input_dtype` here to `input_dtype_with_constraints` to be consistent, same for other occurences